common4j/src/main/com/microsoft/identity/common/java/authorities/CIAMAuthority.java

```diff
@@ -0,0 +1,96 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.java.authorities;
+
+import com.microsoft.identity.common.java.exception.ClientException;
+import com.microsoft.identity.common.java.logging.Logger;
+import com.microsoft.identity.common.java.providers.microsoft.azureactivedirectory.AzureActiveDirectorySlice;
+import com.microsoft.identity.common.java.providers.microsoft.microsoftsts.MicrosoftStsOAuth2Configuration;
+import com.microsoft.identity.common.java.providers.microsoft.microsoftsts.MicrosoftStsOAuth2Strategy;
+import com.microsoft.identity.common.java.providers.oauth2.OAuth2Strategy;
+import com.microsoft.identity.common.java.providers.oauth2.OAuth2StrategyParameters;
+
+import lombok.NonNull;
+
+public class CIAMAuthority extends Authority {
+    private static transient final String TAG = CIAMAuthority.class.getSimpleName();
+
+    // Adding this flag to indicate whether or not we should load the OpenId Configuration as part of handling
+    // CIAM flows. This is currently relevant for fetching the authorization endpoint from OpenId rather than
+    // using the authority itself + adding the default authorization endpoint.
+    private final boolean CIAM_USE_OPENID_CONFIGURATION = true;
```

Comment:
This sounds like the right approach. It is usually best to load them from Open Id.

Reply:
We do not currently do this for AAD, we can create some work in the future to update AAD to also use openid configuration?

Reply:
Hmm. Interesting. Perhaps there are reasons for not doing it historically. Probably a good idea to create a work item to revisit this at some point and see if we can leverage Open Id Configuration in all places.

Reply:
Sounds good!

Reply:
Not much bandwidth this quarter, lets rain check this